packages/@aws-cdk/aws-iam/lib/instance-profile-ref.ts

```diff
@@ -0,0 +1,95 @@
+import cdk = require('@aws-cdk/cdk');
+import { PolicyStatement } from './policy-document';
+import { Role } from './role';
+
+export interface InstanceProfileRefProps {
+
+    /**
+     * The path associated with this instance profile. For information about IAM Instance Profiles, see
+     * Friendly Names and Paths in IAM User Guide.
+     */
+    path: string;
+
+    /**
+     * The name of an existing IAM role to associate with this instance profile.
+     * Currently, you can assign a maximum of one role to an instance profile.
+     * @default Creates a default ec2.amazonaws.com Assume Role.
+     */
+    role?: Role;
```

Comment:
This should probably be an `IRole`.

Reply:
@rix0rrr The only issue with making it an `IRole` is that the `roleName` is not an attribute of `IRole`.  The `roleName` is used on the `InstanceProfile` as the reference to the `Role`.  

I see 3 options to solve this:  

1. Keep `Role`
2. Use `IRole`, generate a name if the role instance does not support `roleName`
3. Add `roleName` to `IRole`

Reply:
Totally 3!

Reply:
cool.  Just wanted to make sure before I started adding stuff.